pyccel/ast/type_annotations.py

```diff
@@ -205,6 +218,20 @@ def results(self):
     def __repr__(self):
         return f'func({repr(self.args)}) -> {repr(self.results)}'
 
+    @property
+    def is_const(self):
+        """
+        Indicates whether the object will remain constant.
+
+        Returns a boolean which is false if the value of the object can be
+        modified, and true otherwise.
+        """
+        return self._is_const
+
+    @is_const.setter
+    def is_const(self, val):
+        self._is_const = val
```

Comment:
Same comment as above about checking the type

Reply:
Fixed f6c7ee3e